leader.go

```diff
@@ -262,39 +262,52 @@ func (k *Kinsumer) registerLeadership() (bool, error) {
 }
 
 // loadShardIDsFromKinesis returns a sorted slice of shardIDs from kinesis.
-// This function uses kinesis.DescribeStream, which has a very low throttling limit of 10/s per account.
-// To avoid hitting that limit, unless you need an as-recent-as-possible list,
-// you should use the cache, returned by loadShardIDsFromDynamo below.
+// This function used to use kinesis.DescribeStream, which has a very low throttling limit of 10/s per account.
+// As such, the leader is responsible for caching the shard list.
+// Now that it uses ListShards, you could potentially query the shard list directly from all clients.
```

Comment:
Should we consider making this change?

Reply:
https://github.com/twitchscience/kinsumer/issues/42 relevant right?

Reply:
Yeah, I think it's worth making. The ListShards limit is 100/second/stream, which seems like plenty. It's then worth considering if the leader can be removed (I think it still handles client lists to limit Dynamo usage, but I'm not sure if that's necessary).
#42 is a little relevant in that it's a per-consumer hit of DescribeStream. We actually hit that same issue recently and I was considering just removing the healthiness check since the reads will just fail if the stream isn't healthy anyway.